src/command_modules/azure-cli-appservice/azure/cli/command_modules/appservice/custom.py

```diff
@@ -529,21 +529,42 @@ def create_webapp_slot(resource_group_name, webapp, slot, configuration_source=N
     return result
 
 
-def config_source_control(resource_group_name, name, repo_url, repository_type=None, branch=None,  # pylint: disable=too-many-locals
-                          git_token=None, manual_integration=None, slot=None, cd_provider=None,
-                          cd_app_type=None, cd_account=None, cd_account_must_exist=None):
+def config_source_control(resource_group_name, name, repo_url, repository_type='git', branch=None,  # pylint: disable=too-many-locals
+                          manual_integration=None, git_token=None, slot=None, cd_app_type=None,
+                          app_working_dir=None, nodejs_task_runner=None, python_framework=None,
+                          python_version=None, cd_account_create=None, cd_project_url=None, test=None,
+                          slot_swap=None, private_repo_username=None, private_repo_password=None):
     client = web_client_factory()
     location = _get_location_from_webapp(client, resource_group_name, name)
 
-    if cd_provider == 'vsts':
-        create_account = not cd_account_must_exist
+    if cd_project_url:
+        # Add default values
+        cd_app_type = 'AspNet' if cd_app_type is None else cd_app_type
+        python_framework = 'Django' if python_framework is None else python_framework
+        python_version = 'Python 3.5.3 x86' if python_version is None else python_version
```

Comment:
Can we avoid hardcode these values in the CLI?

Reply:
Default values need to be hardcoded somewhere, if not here then may be in some separate file to manage constants (harcoded values).

Reply:
Does service has endpoint provides default value? How does python SDK user get the default value?

Reply:
Service doesn't provide any default values. SDK users will get to know about default values via command arguments help text.
I can't see any other way of doing this because we need to set default values only if a particular argument (cd_project_url) is set. That is the reason I've removed default values from command signature function.
Let me know if you can suggest any other way of doing this.